src/pasteup/grid.js

```diff
@@ -47,17 +47,27 @@ export const calculateWidth = (breakpoint, colspan) => {
     );
 };
 
-const gridStyles = (breakpoint, [colspan]) => ({
-    [breakpointMqs[breakpoint]]: {
-        float: 'left',
-        width: calculateWidth(breakpoint, colspan) + gutter,
-        paddingLeft: gutter,
-    },
-});
+const gridStyles = (breakpoint, [colspan]) => {
+    let basis;
+
+    if (typeof colspan === 'number') {
+        basis = `${calculateWidth(breakpoint, colspan) + gutter}px`;
+    } else if (colspan === 'auto') {
+        basis = 'auto';
+    }
+
+    return {
+        [breakpointMqs[breakpoint]]: {
+            flex: `0 1 ${basis}`,
+            paddingLeft: gutter,
+        },
+    };
+};
 
 const RowStyled = styled('div')({
     ...clearFix,
     marginLeft: -gutter,
+    display: 'flex',
```

Comment:
should we fallback here to every col 100% wide with no flex?

Reply:
actually, we know the offset and the max columns, can we set the width the remainder and avoid fallback?